portlist/poller.go

```diff
@@ -137,6 +162,27 @@ func (p *Poller) send(ctx context.Context, pl List) (sent bool, err error) {
 	}
 }
 
+// Poll synchronously polls for changes in the ports.
+// Note that the return ports is a total list of open ports,
+// but is nil when there are no changes. The caller will
+func (p *Poller) Poll() (ports []Port, changed bool, err error) {
+	p.initOnce.Do(func() {
+		p.initErr = p.init()
+	})
```

Comment:
@bradfitz this _will_ be the only place it is called from once we remove the async API. 

Fun fact, the `getList` method does this init-ing purely for testing purposes, so I look forward to removing the need for that. 

In the meantime should we:

1. Keep things as they are?
2. Just add a todo comment for the follow up PR?
3. Consolidate all the soon-to-be-removed once calls to use the same init function?

Reply:
Fix the init story now. Because it's already broken. Add this test to see:

```go
func TestClose(t *testing.T) {
        var p Poller
        p.Close()
}
```

And watch it explode with this PR.

Reply:
@bradfitz since there is no constructor, we can't guarantee a user doesn't call Close before calling Poll, so the Close should just be protected anyway (which is fixed in the latest commit) -- so it's somewhat unrelated to different functions inside the Once. Should I still proceed with option 3?

Reply:
yes, by "fix the init story" I meant (3).

Reply:
@bradfitz pushed an update, all `once` calls are now using the same `init` function.